kotlinx-coroutines-core/jsAndWasmJsShared/src/internal/CoroutineExceptionHandlerImpl.kt

```diff
@@ -0,0 +1,17 @@
+package kotlinx.coroutines.internal
+
+import kotlinx.coroutines.*
+
+internal expect interface JsAny
+
+internal expect fun Throwable.toJsException(): JsAny
+
+/*
+ * Schedule an exception to be thrown inside JS or Wasm/JS event loop,
+ * rather than in the current execution branch.
+ */
+internal fun throwAsync(e: JsAny): Unit = js("setTimeout(function () { throw e }, 0)")
+
+internal actual fun propagateExceptionFinalResort(exception: Throwable) {
+    throwAsync(exception.toJsException())
```

Comment:
@murfel @dkhalanskyjb can we throw an exception right from here? May it break anything inside kx-coroutines? 

In my demo project, `throwAsync` was mainly needed to escape kx-coroutines handling and emulate the unhandled exception behavior while using `CoroutineExceptionHandler`.